README.md

```diff
@@ -72,6 +72,17 @@ remove_exclude_constraint :booking
 remove_exclude_constraint :booking, using: :gist, 'tsrange("from", "to")' => :overlaps, room_id: :equals
 ```
 
+#### Constrainsts with custom conditions
+
+if you need custom/complex conditions of any sort, you can even use string conditions:
+
+```ruby
+create_table :people do |t|
+  t.string :title
+  t.check_constraint "title IS NULL OR title IN ['Mr.', 'Mrs.', 'Dr.']"
```

Comment:
This is an interesting use-case.  To be more consistent with how rails translates nil into NULL (i.e. `MyModel.where(title: [nil, 'Mr.', 'Mrs.', 'Dr.'])`, we should really make it possible to use `t.check_constraint title: [nil, 'Mr.', 'Mrs.', 'Dr.']`.

I'm fine with leaving this example as is for now.  Just mentioning the idea so that I can cycle back to it at some point in the future.

Reply:
Lets put this in an issue so this can be tracked and maybe some contributor can pick it.